tests/at_functional_test/test/atclient_sharedkey_test.dart

```diff
@@ -41,17 +40,31 @@ void main() {
       ..sharedBy = currentAtSign
       ..metadata = (Metadata()..ttl = 120000);
     var value = '+91 887 888 3435';
-    var encryptionService = AtKeyEncryptionManager(atClient)
-        .get(phoneKey, currentAtSign);
+    var encryptionService =
+        AtKeyEncryptionManager(atClient).get(phoneKey, currentAtSign);
     var encryptedValue = await encryptionService.encrypt(phoneKey, value);
     var result = await atClient.getRemoteSecondary()!.executeCommand(
         'update:sharedKeyEnc:${phoneKey.metadata?.sharedKeyEnc}:pubKeyCS:${phoneKey.metadata?.pubKeyCS}:${phoneKey.sharedWith}:${phoneKey.key}.$namespace$currentAtSign $encryptedValue\n',
         auth: true);
     expect(result != null, true);
-    await FunctionalTestSyncService.getInstance().syncData(atClientManager.atClient.syncService);
-    var metadata = await atClient.getMeta(phoneKey);
-    expect(metadata?.sharedKeyEnc, isNotEmpty);
-    expect(metadata?.pubKeyCS, isNotEmpty);
+    // sync to local
+    await FunctionalTestSyncService.getInstance()
+        .syncData(atClientManager.atClient.syncService);
+    // check if local and remote are in sync before assertions
+    bool isInSync = await atClientManager.atClient.syncService.isInSync();
```

Comment:
This will not completely fix the issue. If for some reason, sync takes 3 iterations to sync the expected key, the assertation will be skipped.